### PR TITLE
Add GitHub Action to Automatically Create Releases

### PR DESCRIPTION
**Pull request by adisidev**
Many thanks to @Wind1337. 

This was previously opened as #201 , however, that had breaking changes on macOS. Thus, I've made the new CMakeLists.txt work on macOS and merged the previous changes from Carlsen, retaining the correct commit history so that Carlsen can be correctly credited.

### CODE REVIEW

File: src/cartogram_info/cartogram_info.cpp [hunk 1]
```diff
@@ -204,3 +205,36 @@ std::string CartogramInfo::set_map_name(const std::string &map_name)
   }
   return map_name_;
 }
+
+void CartogramInfo::write_csv(const std::string &csv_file_name) {
+  // Write a csv file with the current target areas
```
Comment (nihalzp) [inline src/cartogram_info/cartogram_info.cpp:210]:
```suggestion
    // Write a csv file with the current target areas
```

File: src/cartogram_info/cartogram_info.cpp [hunk 2]
```diff
@@ -204,3 +205,36 @@ std::string CartogramInfo::set_map_name(const std::string &map_name)
...
+    std::ofstream out_file_csv;
+    out_file_csv.open(csv_file_name + ".csv");
+    if (!out_file_csv) {
+      std::cerr
+        << "ERROR writing GeoJSON: failed to open " << csv_file_name << ".csv"
```
Comment (nihalzp) [inline src/cartogram_info/cartogram_info.cpp:215]: Is this message accurate?

File: src/cartogram_info/write_geojson.cpp [hunk 3]
```diff
@@ -193,7 +193,7 @@ void CartogramInfo::write_geojson(
     combined_json["Original"] = new_json_original;
     std::cout << combined_json << std::endl;
```
Comment (nihalzp) [inline src/cartogram_info/write_geojson.cpp:194]: Should we also include the `equal-area-map` as we would require it in the future on the website?

File: src/inset_state/inset_state.cpp [hunk 4]
```diff
@@ -182,7 +182,7 @@ bool InsetState::insert_constraint_safely(const Point &p1, const Point &p2)
     proj_qd_.dt.insert_constraint(p1, p2);
     return true;
   } catch (const std::exception &e) {
-    std::cout << "WARNING DIAGONAL: Could not insert constraint between " << p1
+    std::cerr << "WARNING DIAGONAL: Could not insert constraint between " << p1
```
Comment (nihalzp) [inline src/inset_state/inset_state.cpp:185]: Good catch!

**Review by nihalzp — DISMISSED** [review]

File: src/cartogram_info/cartogram_info.cpp (hunk 2 above)
Comment (adisidev) [reply src/cartogram_info/cartogram_info.cpp:215]: Haha, that is a good question. Changed!

File: src/cartogram_info/write_geojson.cpp (hunk 3 above)
Comment (adisidev) [reply src/cartogram_info/write_geojson.cpp:194]: In a future pull-request, @atima can tack that up as an issue when needed.

File: src/inset_state/inset_state.cpp (hunk 4 above)
Comment (adisidev) [reply src/inset_state/inset_state.cpp:185]: Learnt from the best!

**Review by nihalzp — APPROVED** [review]

**Pull request merged**